### PR TITLE
Refactor to centralize duplicate TopologyAPI.Config logic

### PR DESCRIPTION
**Pull request by billonahill**
This inconsistency bit me during hackweek. Refactoring to share duplicate config handling logic.

### CODE REVIEW

**Comment by billonahill** [comment]
Ping. Any takers?

File: heron/api/src/java/com/twitter/heron/api/utils/Utils.java [hunk 1]
```diff
@@ -102,4 +112,39 @@ public static Object deserialize(byte[] serialized) {
...
+   * keys or values are ignored.
+   *
+   * @param config heron Config object
+   * @return TopologyAPI.Config.Builder with values loaded from config
+   */
+  public static TopologyAPI.Config.Builder getConfigBuilder(Config config) {
+    TopologyAPI.Config.Builder cBldr = TopologyAPI.Config.newBuilder();
+    Set<String> apiVars = config.getApiVars();
+    for (String key : config.keySet()) {
+      Object value = config.get(key);
```
Comment (objmagic) [inline heron/api/src/java/com/twitter/heron/api/utils/Utils.java]: shall we check if `key` is null before fetching value?

**Comment by objmagic** [comment]
LGTM except a minor comment.

File: heron/api/src/java/com/twitter/heron/api/utils/Utils.java (hunk 1 above)
Comment (billonahill) [inline heron/api/src/java/com/twitter/heron/api/utils/Utils.java]: Good call, updated.

File: heron/api/src/java/com/twitter/heron/api/utils/Utils.java [hunk 2]
```diff
@@ -25,8 +25,18 @@
...
+import java.util.logging.Logger;
+
+import com.google.protobuf.ByteString;
+
+import com.twitter.heron.api.Config;
+import com.twitter.heron.api.HeronTopology;
+import com.twitter.heron.api.generated.TopologyAPI;
 
 public final class Utils {
+  private static final Logger LOG = Logger.getLogger(HeronTopology.class.getName());
```
Comment (nlu90) [inline heron/api/src/java/com/twitter/heron/api/utils/Utils.java]: why here uses the logger of `HeronTopology`?

Comment (billonahill) [inline heron/api/src/java/com/twitter/heron/api/utils/Utils.java]: Good catch, fixed.

**Comment by nlu90** [comment]
👍

**Pull request merged**